services/meta/client.go

```diff
@@ -694,11 +694,19 @@ func (c *Client) DropShard(id uint64) error {
 
 // CreateShardGroup creates a shard group on a database and policy for a given timestamp.
 func (c *Client) CreateShardGroup(database, policy string, timestamp time.Time) (*ShardGroupInfo, error) {
+	// Check under a read-lock
```

Comment:
`CreateShardGroup` is actually `CreateShardGroupIfNotExists` and shard groups are checked for existence much more frequently than they are created.  The write lock showed up in the block profile when writing points w/ many fields.